### ISSUE TITLE
[STORM-1992] Deploy multilang-javascript code as node package

### ISSUE THREAD

**Issue by jira-importer** [apache/storm#5774]
<p>Now that storm includes Flux, it is easier than ever to deploy a topology with javascript components. If the Bolt and Spout base classes defined in storm/multi-lang/javascript were available in a node.js package on <a href="https://www.npmjs.com/" class="external-link" target="_blank" rel="nofollow noopener">https://www.npmjs.com/</a> it would allow node.js storm users to take advantage of node's built in package manager to develop their own bolts and spouts.</p>

<p>It would be relatively trivial to add some maven tasks to storm/multi-lang/javascript/pom.xml to take the storm.js resource and package it in a node module and submit it to npm.</p>

<p>This could be added to the pom as a separate profile so it wouldn't impact the normal storm build process.</p>

<p>This integration will also make it easier to add unit tests for storm.js</p>

<p>For additional background see this discussion: </p>

<p><a href="http://mail-archives.apache.org/mod_mbox/storm-dev/201607.mbox/%3CCAD8EKPHc6O1LCnoQUUoYoDuMQ3uSaNpD5gR4onK%2B0EL5_qcZ3Q%40mail.gmail.com%3E" class="external-link" target="_blank" rel="nofollow noopener">http://mail-archives.apache.org/mod_mbox/storm-dev/201607.mbox/%3CCAD8EKPHc6O1LCnoQUUoYoDuMQ3uSaNpD5gR4onK%2B0EL5_qcZ3Q%40mail.gmail.com%3E</a></p>

<p>If this sounds like a worthwhile addition to the project I would be happy to submit a PR.</p>

---
<details><summary><i>Originally reported by <a title="Marc Zbyszynski" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=marrrck">marrrck</a>, imported from: <a href="https://issues.apache.org/jira/browse/STORM-1992" target="_blank">Deploy multilang-javascript code as node package</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2024-07-02T10:45:13+01:00
<li><b>imported</b>: 2025-01-24
</ul></i>
</details>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>GitHub user mzbyszynski opened a pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/1584" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/1584</a></p>

<p>    <a href="https://issues.apache.org/jira/browse/STORM-1992" title="Deploy multilang-javascript code as node package" class="issue-link" data-issue-key="STORM-1992"><del>STORM-1992</del></a> Added artifacts to make storm.js into an npm package</p>

<p>    Posting for feedback! Please let me know if I am on the right track. This PR relates to <a href="https://issues.apache.org/jira/browse/STORM-1992" title="Deploy multilang-javascript code as node package" class="issue-link" data-issue-key="STORM-1992"><del>STORM-1992</del></a>(<a href="https://issues.apache.org/jira/browse/STORM-1992" class="external-link" rel="nofollow">https://issues.apache.org/jira/browse/STORM-1992</a>)</p>

<p>    ```bash<br/>
    $ mvn deploy -P npmjs<br/>
    ```<br/>
    Would publish the package to npmjs.com as *<b>storm-multilang-js</b>* (I could not think of a better name that was not taken) along with the other deployments the build performs. It would have be manually executed right now because you need to have run `npm login` before hand to enter your credentials. I have not found a clean way to make this automated, but if that is a requirement I can probably get it working in an unclean way.</p>

<p>    I published the package with a bogus version number here as an example: <a href="https://www.npmjs.com/package/storm-multilang-js" class="external-link" target="_blank" rel="nofollow noopener">https://www.npmjs.com/package/storm-multilang-js</a></p>

<p>    I added a `stormjs` utility to make it easier to call a js spout or bolt from a topology by taking care of some of the boilerplate. I also cleaned up `storm.js` a little bit by adding missing semi-colons and making string quotation consistent.</p>

<p>    I'm unsure about the README.md file for the npm package. Should I add more to it or should it just point to documentation living somewhere else?</p>

<p>    Thank you!</p>

<p>You can merge this pull request into a Git repository by running:</p>

<p>    $ git pull <a href="https://github.com/mzbyszynski/storm" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/mzbyszynski/storm</a> <a href="https://issues.apache.org/jira/browse/STORM-1992" title="Deploy multilang-javascript code as node package" class="issue-link" data-issue-key="STORM-1992"><del>STORM-1992</del></a>_multilang_js_npm</p>

<p>Alternatively you can review and apply these changes as the patch at:</p>

<p>    <a href="https://github.com/apache/storm/pull/1584.patch" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/1584.patch</a></p>

<p>To close this pull request, make a commit to your master/trunk branch<br/>
with (at least) the following in the commit message:</p>

<p>    This closes #1584</p>

<hr />
<p>commit afe5e3e4db519ea5877c6e9a7c5b2da668784814<br/>
Author: Marc Zbyszynski <mzbyszynski@xlhybrids.com><br/>
Date:   2016-07-21T21:55:57Z</p>

<p>    <a href="https://issues.apache.org/jira/browse/STORM-1992" title="Deploy multilang-javascript code as node package" class="issue-link" data-issue-key="STORM-1992"><del>STORM-1992</del></a> Added artifacts to multilang-javascript to package it as an npm module, and added npmjs profile for deploying the package to npmjs.con</p>

<hr />

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rzo1">rzo1</a>:</i>
<p>Storm 1.x isn't maintained anymore.</p>
